From 6e82d0a8690b08b3420e5a4a79aef872a94ac1a7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lisandro=20Dami=C3=A1n=20Nicanor=20P=C3=A9rez=20Meyer?=
 <perezmeyer@gmail.com>
Date: Mon, 9 Sep 2019 18:31:03 -0300
Subject: [PATCH] ClangFormat: improve the message the users see

In this way it's made clear that the plugin requires a patched version of Clang.

Task-number: QTCREATORBUG-22923
Change-Id: Ifefd5ab1a3dc82b85e553c5e1fedffb72919dd17
---
 src/plugins/clangformat/clangformatplugin.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/plugins/clangformat/clangformatplugin.cpp b/src/plugins/clangformat/clangformatplugin.cpp
index 49f4fe049f..f6853ae836 100644
--- a/src/plugins/clangformat/clangformatplugin.cpp
+++ b/src/plugins/clangformat/clangformatplugin.cpp
@@ -154,8 +154,9 @@ bool ClangFormatPlugin::initialize(const QStringList &arguments, QString *errorS
     }
     return true;
 #else
-    *errorString = "Disabling ClangFormat plugin as it is not built against a suitable version of "
-                   "Clang's libFormat. For more information, see the Qt Creator README at "
+#warning ClangFormat: building dummy plugin due to unmodified Clang, see README.md for more info
+    *errorString = "Disabling ClangFormat plugin as it has not been built against a modified Clang's libFormat."
+                   "For more information see the Qt Creator README at "
                    "https://code.qt.io/cgit/qt-creator/qt-creator.git/tree/README.md";
     return false;
 #endif
-- 
2.23.0

